fix: header styles and heading of projects

aqsaaqeel opened the pull request:
## Summary

-> Used the finalized design in https://github.com/unstructuredstudio/zubhub/issues/792 to develop the hero section. 

-> The hero table is not populated so the image needs to be added from the backend. 

-> To run it and test, I added the Hero details on [http://localhost:8000/admin](https://github.com/unstructuredstudio/zubhub/pull/url)

-> As for this PR, I have made all the necessary CSS changes and added the necessary components.


## Closes #792 

## Changes

- Hero Container design on landing page
- Translated title for both English and Hindi
- Font Size of the heading Featured Project to match the rest

## Screenshots
I have made the changes according to @srish instructions [here](https://github.com/unstructuredstudio/zubhub/issues/792#issuecomment-1756589431)
![Screenshot from 2023-10-12 18-43-00](https://github.com/unstructuredstudio/zubhub/assets/85514520/dec93186-0038-47d2-a582-c2a817fe1e28)

Commit:
fix: header styles and heading of projects

Commit:
test commit

Commit:
revert: changes made to Featured Projects

Commit:
fix: hero title updated

Commit:
style: changed styles for hero section on homepage

Commit:
style: adjusted button size

aqsaaqeel commented:
I have made the changes you suggested @tuxology SInce the hero section text had a different style than the rest of the titles, they needed new classes for styles to be made. Here is how the hero section looks now.
![Screenshot from 2023-10-13 10-44-13](https://github.com/unstructuredstudio/zubhub/assets/85514520/c5f392f9-89ca-4b5d-92a6-ff322b53b063)
![Screenshot from 2023-10-13 10-43-38](https://github.com/unstructuredstudio/zubhub/assets/85514520/df37cabc-b986-47ed-bd54-f92898215c62)

tuxology commented on zubhub_frontend/zubhub/src/views/home/Home.jsx:
```diff
@@ -103,8 +102,8 @@ function Projects(props) {
               {hero && hero.id ? (
-                 <Box className={classes.heroSectionStyle}>
-                 <Box className={classes.heroContainerStyle}>
+                 <Box className={classes.heroMainSectionStyle}>
```
Why have we renamed it to `heroMainSection` style instead of old name?

aqsaaqeel replied:
There is another section on this page that uses the ```heroSectionStyle``` name for its classes. It is the carousel section for ambassadors and their testimonials. 
Making any changes to the already existing classes meant changing other sections as collateral too. However, since we were focusing on just the Hero Section and it required different styling than the rest of the components, I made a new class.

tuxology reviewed (changes requested):
Few changes we need: 
 - [ ]  The UI does not match the design in terms of width of hero element. It is currently full width however in the proposed design, we need to align it in the same width as the container containing elements of navbar. 
 -  [ ] In mobile view, there needs to be padding between Share your Work button and the image and the image needs to be center aligned

Commit:
fix: navbar padding

aqsaaqeel commented:
> Few changes we need:
> 
> * [ ]   The UI does not match the design in terms of width of hero element. It is currently full width however in the proposed design, we need to align it in the same width as the container containing elements of navbar.
> * [ ]  In mobile view, there needs to be padding between Share your Work button and the image and the image needs to be center aligned

Changes I have made in this commit:
- The padding for `navbar` is adjusted.
- Padding space between `hero button` and `hero image` has been added.

![Screenshot from 2023-10-14 05-54-03](https://github.com/unstructuredstudio/zubhub/assets/85514520/595839bc-e179-49e0-b690-968c938881ee)
As for the image on mobile screen, it looks centered here, is this what you meant? @tuxology 
![Screenshot from 2023-10-14 05-51-56](https://github.com/unstructuredstudio/zubhub/assets/85514520/f281f775-d7fc-4be2-9100-4982f66e066f)


Since the scope of this PR is just the hero section, I haven't altered much on the navbar, but I feel like we should also implement the design for the icons from [this](https://www.figma.com/file/nyUduEpFmVZ05xA8rt28Mt/ZubHub-Feature-Re-Designs?type=design&node-id=11-414&mode=design&t=xDeX7LSKbik6YPFy-0)

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/views/projects/projectsStyles.js:
```diff
@@ -108,20 +137,44 @@ const styles = theme => ({
     marginTop: '1.2em',
+    width: 'fit-content',
+    height: '3.95rem',
+    padding: '1.04rem 2.08rem 1.04rem 2.08rem',
```
So there are two decimal places here. Why do we need such accuracy?

aqsaaqeel replied:
I use the Figma design to show me the exact styles applied, and those are in `px`. I convert them to `rem` because it is more flexible in terms of adjusting according to screen sizes. The `px` values are a whole number but converting them to `rem` gives these decimal values.

tuxology replied:
@aqsaaqeel Can we round the double digit values to single digit at the very minimum? You can eyeball them to see if they align. Usually, there should be minimum need to do such "fine tuning". The figma designs are not supposed to be accurate representations to the exact location, but just indication of how things are aligned. Another benefit of keeping simple numbers is that we don't have to do a lot of fine tuning in other places (containers/padding) - since they will also align to single values.

aqsaaqeel replied:
Alright. I'll make the changes!

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/views/projects/projectsStyles.js:
```diff
@@ -108,20 +137,44 @@ const styles = theme => ({
     textTransform: 'none',
-    padding: '0 1em',
-    fontSize: '1.2rem',
+    fontSize: '1.456rem',
```
Why do we need 3 decimal accurate values? How have these been calculated? I feel if we need such fine adjustments, then there must be some underlying issue

aqsaaqeel replied:
No, I don't think there is any underlying issue. My reasoning is the same as above ☝️

tuxology reviewed (approved):
I am fine with this, but need a few clarifications/modifications on comments before we merge @aqsaaqeel

Commit:
fix: rounded off decimals in styles

Commit:
fix: rounded off decimals in styles

tuxology commented:
@aqsaaqeel Can you revert navbar height? We don't want to change it yet

tuxology reviewed (changes requested):
See comment about navbar height.

Commit:
revert Navbar changes

aqsaaqeel commented:
> See comment about navbar height.

Alright! Can I work on #801 too on this PR? @tuxology

tuxology commented:
@aqsaaqeel Lets leave #801 for now.

Also, sorry for increasing the scope of this PR, but can we actually harcode the header SVG here in the code itself? (add the  SVG to the PR and use it). We can keep the custom image upload feature (which is currently in Django Admin) still. We can add a checkbox in the Django Admin to select default image to be used OR uploaded image. Is that doable?

aqsaaqeel commented:
> We can add a checkbox in the Django Admin to select default image to be used OR uploaded image

It's alright. I'm on it!

Commit:
fix: hero image hardcoded

aqsaaqeel commented:
@tuxology I've added the svg image but I am not able to understand why we need the optional image on admin. We are controlling what goes in the header image in the frontend anyway. So wouldn't the choice of which image we want to display come from the frontend even if we want to put the image from Admin? 🤔

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/views/projects/projectsStyles.js:
```diff
@@ -25,20 +25,35 @@ const styles = theme => ({
-    MozBoxShadow: '-2px 25px 13px -18px rgba(220,53,69,0.7)',
     display: 'flex',
     paddingTop: '2.3em',
     paddingBottom: '2.85em',
```
Can you do rounding?

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/views/projects/projectsStyles.js:
```diff
@@ -99,6 +127,7 @@ const styles = theme => ({
   heroMessageContainerStyle: {
     flexGrow: 1,
     color: 'white',
+    maxWidth: '37.813rem',
```
Rounding

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/views/projects/projectsStyles.js:
```diff
@@ -108,20 +137,44 @@ const styles = theme => ({
     marginRight: '1em',
     marginTop: '1.2em',
+    width: 'fit-content',
+    height: '3.95rem',
```
Rounding

tuxology reviewed (changes requested):
Please review the code again for decimal rounding suggestions. Eyeball the UI once you have rounded the values.

tuxology commented:
> @tuxology I've added the svg image but I am not able to understand why we need the optional image on admin. We are controlling what goes in the header image in the frontend anyway. So wouldn't the choice of which image we want to display come from the frontend even if we want to put the image from Admin? 🤔

The idea for user provided hero element was to make the image modifiable from time to time (eg. highlight some news, upcoming event etc). The flow is this:
 - Default image is one hardcoded in frontend. In absence of any header image upload from Django Admin, this is what is shown
 - If custom image is uploaded AND the checkbox ticked, then the custom image takes precedence in rendering. 
 - If custom image is uploaded but the checkbox is not ticked, default image will be used

This flow can be coded in the frontend to get this working.

Commit:
fix: show hero on condition

aqsaaqeel commented:
> > @tuxology I've added the svg image but I am not able to understand why we need the optional image on admin. We are controlling what goes in the header image in the frontend anyway. So wouldn't the choice of which image we want to display come from the frontend even if we want to put the image from Admin? 🤔
> 
> The idea for user provided hero element was to make the image modifiable from time to time (eg. highlight some news, upcoming event etc). The flow is this:
> 
> * Default image is one hardcoded in frontend. In absence of any header image upload from Django Admin, this is what is shown
> * If custom image is uploaded AND the checkbox ticked, then the custom image takes precedence in rendering.
> * If custom image is uploaded but the checkbox is not ticked, default image will be used
> 
> This flow can be coded in the frontend to get this working.

I have added the conditional to render the hardcoded image if there isn't one coming from the admin already.
As per the use case which you have mentioned, when we want a modified image, we can:
-> Go to admin panel.
-> Upload custom image.
After the said event expires, we can remove the custom image.
-> To clear it, there's already a clear selection box given. We can clear, save changes. And refresh.

![Untitled design (1)](https://github.com/unstructuredstudio/zubhub/assets/85514520/06deea7c-b982-4eb7-97dc-19a9ea459d34)



Do we have to add a checkbox on top of this ☝️
 
There is another issue that we can see now. The image coming from the admin works properly but the concern is the design does not have padding on the right as we can see here. (this is not an issue for the hardcoded image because of how it is designed) Should I work on this?

![Screenshot from 2023-10-20 00-23-55](https://github.com/unstructuredstudio/zubhub/assets/85514520/8cc3e35f-a1ef-47be-ad29-08fc51b395c6)

tuxology commented:
I think this is fine for now 👍 We can iteratively improve it further

tuxology reviewed (approved):
LGTM 🌈
